lib/rucio/web/rest/webpy/v1/authentication.py

```diff
@@ -441,6 +450,103 @@ def GET(self):
+    """
+
+    def OPTIONS(self):
+        """
+        HTTP Success:
+            200 OK
+
+        Allow cross-site scripting. Explicit for Authentication.
+        """
+
+        header('Access-Control-Allow-Origin', ctx.env.get('HTTP_ORIGIN'))
+        header('Access-Control-Allow-Headers', ctx.env.get('HTTP_ACCESS_CONTROL_REQUEST_HEADERS'))
+        header('Access-Control-Allow-Methods', '*')
+        header('Access-Control-Allow-Credentials', 'true')
+        header('Access-Control-Expose-Headers', 'X-Rucio-Auth-Token')
+        raise OK
+
+    @check_accept_header_wrapper(['application/octet-stream'])
+    def GET(self):
+        """
+        HTTP Success:
+            200 OK
+
+        HTTP Error:
+            401 Unauthorized
+
+        :param Rucio-Account: Account identifier as a string.
+        :param Rucio-Username: Username as a string.
+        :param Rucio-Password: Password as a string.
+        :param Rucio-AppID: Application identifier as a string.
+        :returns: "Rucio-Auth-Token" as a variable-length string header.
+        """
+
+        header('Access-Control-Allow-Origin', ctx.env.get('HTTP_ORIGIN'))
+        header('Access-Control-Allow-Headers', ctx.env.get('HTTP_ACCESS_CONTROL_REQUEST_HEADERS'))
+        header('Access-Control-Allow-Methods', '*')
+        header('Access-Control-Allow-Credentials', 'true')
+        header('Access-Control-Expose-Headers', 'X-Rucio-Auth-Token')
+
+        header('Content-Type', 'application/octet-stream')
+        header('Cache-Control', 'no-cache, no-store, max-age=0, must-revalidate')
+        header('Cache-Control', 'post-check=0, pre-check=0', False)
+        header('Pragma', 'no-cache')
+
+        saml_nameid = cookies().get('saml-nameid')
+        account = ctx.env.get('HTTP_X_RUCIO_ACCOUNT')
+        appid = ctx.env.get('HTTP_X_RUCIO_APPID')
+        if appid is None:
+            appid = 'unknown'
+        ip = ctx.env.get('HTTP_X_FORWARDED_FOR')
+        if ip is None:
+            ip = ctx.ip
+
+        if saml_nameid:
+            try:
+                result = get_auth_token_saml(account, saml_nameid, appid, ip)
+            except AccessDenied:
+                raise generate_http_error(401, 'CannotAuthenticate', 'Cannot authenticate to account %(account)s with given credentials' % locals())
+            except RucioException as error:
+                raise generate_http_error(500, error.__class__.__name__, error.args[0])
+            except Exception as error:
+                print(format_exc())
+                raise InternalError(error)
+
+            if not result:
+                raise generate_http_error(401, 'CannotAuthenticate', 'Cannot authenticate to account %(account)s with given credentials' % locals())
+
+            header('X-Rucio-Auth-Token', result.token)
+            header('X-Rucio-Auth-Token-Expires', date_to_str(result.expired_at))
+            return str()
+
+        SAML_PATH = '/opt/rucio/lib/rucio/web/ui/common/saml/'
```

Comment:
ok, we can't really depend on a path like this. for what exactly is it needed?

Reply:
This is needed as a pointer to SAML config file (settings.json) and the SP certificate & key.

Reply:
Also, should we change this path or keep it as it is?

Reply:
Let's load the path from the rucio.cfg where you specify it in a `[saml]` section.
Thus: `config_get('saml', 'config')` (or similar)